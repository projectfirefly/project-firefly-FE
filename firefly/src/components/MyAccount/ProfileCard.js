import React, {useContext} from "react";
import "../../styles/MyAccountPage.scss";
import { Link } from "react-router-dom";

import Icon from "../../assets/icons";
import { FaPen } from "react-icons/fa";

//context
import { childContext } from "../../context/ChildProfiles/ChildProfileStore";
import { UPDATE_SELECTED } from "../../context/ChildProfiles/ChildProfileStore";

const ProfileCard = (props) => {

  const [context, dispatch] = useContext(childContext)

  const editButton = async () => {
    console.log(props.id)
    await dispatch({type: UPDATE_SELECTED, payload: props.id})
    props.history.push("/editprofile")
  }

  return (
    <div className="profileCardContainer">
      <div className="edit">
        <p>{props.name}</p>
        <div onClick={editButton} className="iconButton">
          <FaPen />
        </div>
      </div>

      <div className="fireflyIcon">
        <Icon
          name="Firefly"
          style={{
            width: "40%",
          }}
        />
      </div>
    </div>
  );
};

export default ProfileCard;
