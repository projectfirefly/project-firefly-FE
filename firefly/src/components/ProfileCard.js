import React from "react";
import "../styles/MyAccountPage.scss";
import { Link } from "react-router-dom";

import Icon from "../assets/icons";
import { FaPen } from "react-icons/fa";

const ProfileCard = () => {
  return (
    <div className="profileCardContainer">
      <div className="edit">
        <Link to="/editprofile" className="iconButton">
          <FaPen />
        </Link>
      </div>
      <p>Username</p>

      <div className="fireflyIcon">
        <Icon
          name="Firefly"
          style={{
            width: "40%"
          }}
        />
      </div>
    </div>
  );
};

export default ProfileCard;
