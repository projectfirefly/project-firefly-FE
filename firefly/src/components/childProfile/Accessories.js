import React, { useState } from "react";
import Icon from "../../assets/icons";

const svgNames = [
    "",
    "LambdaHatSM",
    "NerdGlassesSM",
    "PinkHeadphoneSM",
    "SunGlassesSM",
];

const pickerStyles = {
    display: "flex",
    justifyContent: "space-between",
    alignItems: "center",
    minHeight: "100px",
};

const Accessories = props => {

    // curSVGindex: 0,
    const [curIndex, setCurIndex] = useState(props.accessory);

    const previous = () => {
        const lastIndex = svgNames.length - 1;
        const shouldResetIndex = curIndex === 0;
        const index = shouldResetIndex ? lastIndex : curIndex - 1;

        setCurIndex(index);
        props.accessoryChange(index);
    }

    const next = () => {
        const lastIndex = svgNames.length - 1;
        const shouldResetIndex = curIndex === lastIndex;
        const index = shouldResetIndex ? 0 : curIndex + 1;

        setCurIndex(index);
        props.accessoryChange(index);
    }

        return (
            <div className="carousel" style={pickerStyles}>
                <Arrow
                    direction="left"
                    clickFunction={previous}
                    name="LeftArrow"
                />
                <SvgImage iconName={svgNames[curIndex]} />
                <Arrow
                    direction="right"
                    clickFunction={next}
                    name="RightArrow"
                />
            </div>
        );
}

const Arrow = ({ direction, clickFunction, name }) => (
    <div className={`slide-arrow ${direction}`} onClick={clickFunction}>
        <Icon name={name} height="60px" />
    </div>
);

const SvgImage = ({ iconName }) => {
    const styles = {
        height: "100px",
    };

    return <Icon name={iconName} className="svg-icon" style={styles} />;
};

export default Accessories;
