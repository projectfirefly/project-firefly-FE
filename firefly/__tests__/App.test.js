import React from 'react';
import Accessories from '../src/components/childProfile/Accessories';
import ProfileCard from '../src/components/MyAccount/ProfileCard';
import {shallow} from 'enzyme';

describe('<App/>', () => {

  it('Verifies nothing was added to Accessories', () => {
    const wrapper = shallow(<Accessories/>)
    expect(wrapper).toMatchSnapshot()
  });

  it('Verifies nothing was added to Profile Card', () => {
    const wrapper = <ProfileCard/>
    expect(wrapper).toMatchSnapshot()
  });

  it('View contents of Profile Card', () => {
    const wrapper = shallow(<ProfileCard name={"test"}/>)
    console.log(wrapper.debug())
  });
})

