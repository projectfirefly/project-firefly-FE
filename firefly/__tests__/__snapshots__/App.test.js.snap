// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> Check Accessories 1`] = `ShallowWrapper {}`;

exports[`<App/> Check Accessories 2`] = `<ProfileCard />`;

exports[`<App/> Verifies nothing was added to Accessories 1`] = `ShallowWrapper {}`;

exports[`<App/> Verifies nothing was added to Profile Card 1`] = `<ProfileCard />`;

exports[`<App/> renders without crashing 1`] = `ShallowWrapper {}`;
